jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/MainFragment.java

```diff
@@ -1,18 +1,71 @@
 package com.e_conomic.jonfirstapp;
 
+import android.app.Activity;
+import android.content.Context;
 import android.os.Bundle;
+import android.support.annotation.Nullable;
 import android.support.v4.app.Fragment;
 import android.view.LayoutInflater;
 import android.view.ViewGroup;
 import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
 
 public class MainFragment extends Fragment {
 
+    // The main activity.
+    MainActivity main_activity;
```

Comment:
What about the fragment references? Say, in `MainActivity.java` should I also write `displayMessageFragment` instead of `display_message_fragment`? :)